client/my-sites/people/delete-user/index.jsx

```diff
@@ -224,7 +224,10 @@ class DeleteUser extends Component {
 		const { translate, isJetpack, siteOwner, user } = this.props;
 
 		// A user should not be able to remove the site owner.
-		if ( ! isJetpack && user.ID === siteOwner ) {
```

Comment:
I think it probably makes sense to have Jetpack continue to mirror core behavior because there isn't a clear concept of 'site owner' for a Jetpack-connnected site.

Reply:
Thanks, I updated PR and aligned the logic with WP Admin.